recipes/hdf5/all/conanfile.py

```diff
@@ -87,6 +93,29 @@ def requirements(self):
         if self.options.parallel:
             self.requires("openmpi/4.1.0")
 
+    def _cmake_new_enough(self, required_version):
+        try:
+            import re
+            from io import StringIO
+            output = StringIO()
+            self.run("cmake --version", output)
+            m = re.search(r"cmake version (\d+\.\d+\.\d+)", output.getvalue())
+            return Version(m.group(1)) >= required_version
+        except:
+            return False
+
+    def _inject_stdlib_flag(self, tc):
+        if self.settings.os == "Linux" and self.settings.compiler == "clang":
+            cpp_stdlib = f" -stdlib={self.settings.compiler.libcxx}".rstrip("1")  # strip 11 from stdlibc++11
+            tc.variables["CMAKE_CXX_FLAGS"] = tc.variables.get("CMAKE_CXX_FLAGS", "") + cpp_stdlib
+
+        return tc
```

Comment:
Is there not a way to patch upstream CMakeLists instead? What kind of weird things are they doing?

Reply:
I am afraid this would require at least 3 different patches. Is it really worth the added complexity?

If we really want to go the patch route, then I think [config/toolchain/clang.cmake](https://github.com/HDFGroup/hdf5/blob/develop/config/toolchain/clang.cmake) is the file to patch.

Unfortunately CMake toolchains are not used in v1.10.5 and earlier, so we would need a different approach to patch older versions.